hal/src/nRF52840/spi_hal.cpp

```diff
@@ -205,210 +274,216 @@ void HAL_SPI_Init(HAL_SPI_Interface spi)
     m_spi_map[spi].bit_order = DEFAULT_BIT_ORDER;
     m_spi_map[spi].data_mode = DEFAULT_DATA_MODE;
     m_spi_map[spi].clock = DEFAULT_SPI_CLOCK;
-    m_spi_map[spi].user_callback = NULL;
+    m_spi_map[spi].spi_ss_state = 0;
+    m_spi_map[spi].spi_dma_user_callback = NULL;
+    m_spi_map[spi].spi_select_user_callback = NULL;
     m_spi_map[spi].transfer_length = 0;
 
     HAL_SPI_Release(spi, nullptr);
 }
 
-void HAL_SPI_Begin(HAL_SPI_Interface spi, uint16_t pin)
-{
+void HAL_SPI_Begin(HAL_SPI_Interface spi, uint16_t pin) {
     // Default to Master mode
     HAL_SPI_Begin_Ext(spi, SPI_MODE_MASTER, pin, NULL);
 }
 
-void HAL_SPI_Begin_Ext(HAL_SPI_Interface spi, SPI_Mode mode, uint16_t pin, void* reserved)
-{
-    if (m_spi_map[spi].ss_pin != NRFX_SPIM_PIN_NOT_USED)
-    {
+void HAL_SPI_Begin_Ext(HAL_SPI_Interface spi, SPI_Mode mode, uint16_t pin, void* reserved) {
+    if (m_spi_map[spi].ss_pin != PIN_INVALID) {
         ss_pin_uninit(spi);
     }
 
-    if (m_spi_map[spi].enabled)
-    {
+    if (m_spi_map[spi].enabled) {
         spi_uninit(spi);
     }
 
-    m_spi_map[spi].ss_pin = pin;
-
-    // TODO: need to support spi slave
+    m_spi_map[spi].ss_pin = (pin == SPI_DEFAULT_SS) ? PIN_INVALID : pin;
```

Comment:
This needs to work correctly. If `SPI_DEFAULT_SS`, the SS pin needs to be set to `SS` for `HAL_SPI_INTERFACE1` and `D5` for `HAL_SPI_INTERFACE2`.

Reply:
Fixed.